### PR TITLE
feat(remote-video-mute): respect remote video mute

### PR DESCRIPTION
**Pull request by chburket**
# COMPLETES
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-408291

[UCF Disable Video Design](https://confluence-eng-gpk2.cisco.com/conf/display/LOCUS/SPARK-308740+In-Meeting+Options+-+video)

## This pull request addresses

We need to respect the disable video option for R6 (the setting of disabled video will come later).

## by making the following changes

A fair amount of the code for remote video muting was already there to match remote audio muting, but there were some missing pieces.  In addition, video is handled a bit differently than audio:

* for audio, remote muted and unmuteAllowed are updated together in locus.self.controls.
* for video, the same information is split between locus.self.controls.video.muted and locus.controls.video.enabled, so I emit SELF_REMOTE_VIDEO_MUTE_STATUS_UPDATED from both places, passing only one of the two values.
* for audio, locus sets remote muted to false on leave meeting (and we update the local mute state)
* for video, locus does not unmute video on leave meeting, so the states must be synced manually on join

You'll see a decent amount of simple reformatting due to Prettier/lint.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

In addition to updating/adding unit tests, I manually tested with the Cantina side of this feature.

### I certified that

- [X] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [X] I discussed changes with code owners prior to submitting this pull request

- [X] I have not skipped any automated checks
- [X] All existing and new tests passed
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/members/util.ts [hunk 1]
```diff
@@ -146,8 +147,9 @@ MembersUtil.generateLowerAllHandsMemberOptions = (requestingParticipantId, locus
 });
 
 MembersUtil.getMuteMemberRequestParams = (options) => {
+  const property = options.isAudio === false ? 'video' : 'audio';
   const body = {
-    audio: {
+    [property]: {
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/members/util.ts:152]: looks good

**Review by arun3528 — APPROVED** [review]
If we have any wiki can we reference them here 

rest looks good

**Review by marcin-bazyl — COMMENTED** [review]
looks good to me, posted just a few minor comments

File: packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js [hunk 2]
```diff
@@ -1,66 +1,152 @@
...
-              });
+        it('rejects when correct display hint is not present for unsetting mute on entry', () => {
+          assert.equal(
+            ControlsOptionsUtil.canUnsetMuteOnEntry(locusInfo.parsedLocus.info.userDisplayHints),
+            false
+          );
+        });
+      });
+
+      describe('canSetDisallowUnmute', () => {
```
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js:55]: This isn't my code.  I guess I need to rebase

File: packages/@webex/plugin-meetings/src/locus-info/controlsUtils.ts [hunk 3]
```diff
@@ -94,6 +98,10 @@ ControlsUtils.getControls = (oldControls: any, newControls: any) => {
       ),
 
       hasBreakoutChanged: !isEqual(previous?.breakout, current?.breakout),
+
+      hasVideoEnabledChanged:
+        newControls.video?.enabled != null &&
```
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/plugin-meetings/src/locus-info/controlsUtils.ts:103]: No, undefined != null is false.  undefined !== null is true.

File: packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js]: Not sure what exactly happened there, but I think I've fixed it.

File: packages/@webex/plugin-meetings/src/locus-info/index.ts [hunk 4]
```diff
@@ -780,6 +788,26 @@ export default class LocusInfo extends EventsScope {
...
+            function: 'updateControls',
+          },
+          LOCUSINFO.EVENTS.CONTROLS_VIDEO_ENABLED_UPDATED,
+          {
+            // muted: not part of locus.controls
+            unmuteAllowed: videoEnabled,
+          }
+        );
+
+        this.updateMeeting({unmuteVideoAllowed: videoEnabled});
```
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/plugin-meetings/src/locus-info/index.ts]: Done!

**Review by marcin-bazyl — APPROVED** [review]

File: packages/@webex/plugin-meetings/src/locus-info/controlsUtils.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/plugin-meetings/src/locus-info/controlsUtils.ts:103]: I changed this to `!== undefined` which is more correct anyway

**Review by arun3528 — APPROVED** [review]

**Pull request merged**